salt/modules/solarisips.py

```diff
@@ -298,39 +303,87 @@ def version(*names, **kwargs):
         salt '*' pkg_resource.version pkg://solaris/entire
 
     '''
-    if names:
-        cmd = ['/bin/pkg', 'list', '-Hv']
-        cmd.extend(names)
-        lines = __salt__['cmd.run_stdout'](cmd).splitlines()
-        ret = {}
-        for line in lines:
-            ret[_ips_get_pkgname(line)] = _ips_get_pkgversion(line)
-        if ret:
-            return ret
-    return ''
+    if len(names) == 0:
+        return ''
+
+    cmd = ['/bin/pkg', 'list', '-Hv']
+    cmd.extend(names)
+    lines = __salt__['cmd.run_stdout'](cmd, ignore_retcode=True).splitlines()
+    ret = {}
+    for line in lines:
+        ret[_ips_get_pkgname(line)] = _ips_get_pkgversion(line)
 
+    # Append package names which are not installed/found
+    unmatched = list(filter(lambda name: not reduce(lambda x, y: x or name in y, ret, False), names))
```

Comment:
The linter is marking `name` as not defined here, but this may be a false positive due to the fact that you are calling a lambda within another lambda.